File: sdks/js/anthropic/cases/2-multi-step.js
/**
 * 2-multi-step: Multi-step Conversation
 *
 * Tests a multi-step conversation with conversation history using Anthropic SDK
 * and verifies that Sentry captures all spans for both API calls.
 */

const { Sentry } = require("../setup");
const Anthropic = require("@anthropic-ai/sdk");
const { runTestCase } = require("../../_test-utils/test-runner.cjs");

async function testLogic(inputs) {
  const { model, system, first_prompt, second_prompt } = inputs;

  // Initialize Anthropic client
  const anthropic = new Anthropic({
    apiKey: process.env.ANTHROPIC_API_KEY,
  });

  // First call
  const firstMessage = await anthropic.messages.create({
    model: model,
    max_tokens: 1024,
    system: system,
    messages: [
      {
        role: "user",
        content: first_prompt,
      },
    ],
  });

  if (!firstMessage.content || firstMessage.content.length === 0) {
    throw new Error("No completion returned from Anthropic (first call)");
  }

  const firstText = firstMessage.content[0].text;

  if (process.env.SENTRY_AI_TEST_VERBOSE === "true") {
    console.log(`    First response: ${firstText}`);
  }

  // Second call with conversation history
  const secondMessage = await anthropic.messages.create({
    model: model,
    max_tokens: 1024,
    system: system,
    messages: [
      {
        role: "user",
        content: first_prompt,
      },
      {
        role: "assistant",
        content: firstText,
      },
      {
        role: "user",
        content: second_prompt,
      },
    ],
  });

  if (!secondMessage.content || secondMessage.content.length === 0) {
    throw new Error("No completion returned from Anthropic (second call)");
  }

  const secondText = secondMessage.content[0].text;

  if (process.env.SENTRY_AI_TEST_VERBOSE === "true") {
    console.log(`    Second response: ${secondText}`);
  }
}

module.exports = runTestCase("2-multi-step", testLogic, Sentry);

File: sdks/py/anthropic/cases/2-multi-step.py
"""
2-multi-step: Multi-step Conversation

Tests a multi-step conversation with conversation history using Anthropic SDK
and verifies that Sentry captures all spans for both API calls.
"""

import os
from anthropic import Anthropic
from test_runner import run_test_case


async def test_logic(inputs):
    """The actual test logic"""
    model = inputs["model"]
    system = inputs["system"]
    first_prompt = inputs["first_prompt"]
    second_prompt = inputs["second_prompt"]

    client = Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))

    # First call
    first_response = client.messages.create(
        model=model,
        system=system,
        messages=[
            {"role": "user", "content": first_prompt}
        ],
        max_tokens=1024,
    )

    first_text = first_response.content[0].text

    if not first_text:
        raise Exception("No output returned from Anthropic (first call)")

    if os.getenv("SENTRY_AI_TEST_VERBOSE") == "true":
        print(f"    First response: {first_text}")

    # Second call with conversation history
    second_response = client.messages.create(
        model=model,
        system=system,
        messages=[
            {"role": "user", "content": first_prompt},
            {"role": "assistant", "content": first_text},
            {"role": "user", "content": second_prompt}
        ],
        max_tokens=1024,
    )

    second_text = second_response.content[0].text

    if not second_text:
        raise Exception("No output returned from Anthropic (second call)")

    if os.getenv("SENTRY_AI_TEST_VERBOSE") == "true":
        print(f"    Second response: {second_text}")

    return second_text


# Export test case functions
test_case = run_test_case("2-multi-step", test_logic)
main = test_case["main"]
assert_sentry = test_case["assert_sentry"]
